### PR TITLE
Feature/pia 1744 improve integration guide

### PR DESCRIPTION
**Pull request by a-szotyori**
Restructure, rewrite and extend the integration guide.

Follow the structure used by the Capture SDK's integration guide and
explain how to use the capture and the payment request handling
features.

For the capture feature also explain how to use the return assistant.

### How to test
View the generated integration guide [here (accessible from the office network)](http://100.105.136.126:8080/job/gini/job/gini-pay-bank-sdk-android/job/feature%252FPIA-1744-improve-integration-guide/Documentation/) (ignore the few styling issues or use the top right "Zip" link to download it and view it locally).

### Ticket 
https://ginis.atlassian.net/browse/PIA-1744

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,61 +6,26 @@ Documentation
...
+Examples
+--------
 
-* Screen API: Android 5.0+ (API Level 21+)
-* Component API: Android 5.0+ (API Level 21+)
+## Invoice Capture
 
-### Phone Hardware
+We are providing example apps for the Screen API and the Component API. These apps demonstrate how
+to integrate the Gini Pay Bank SDK's invoice capture feature:
```
Comment (tastyminerals) [inline README.md:15]: `:` to `.`?

File: ginipaybank/src/doc/source/capture-features.rst [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+            override fun onCancelled() {
+                // Handle cancellation.
+            }
+        })
+    }
+
+Return Assistant
+----------------
+
+The return assistant features allows your users to view and edit the payable items in an invoice. The total amount is
```
Comment (tastyminerals) [inline ginipaybank/src/doc/source/capture-features.rst:122]: `feature`

Comment (tastyminerals) [inline ginipaybank/src/doc/source/capture-features.rst:122]: just `payable items` without `the`

File: ginipaybank/src/doc/source/capture-features.rst [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+    CaptureConfiguration(returnAssistantEnabled = true)
+
+Screen API
+~~~~~~~~~~~
+
+When integrating using the Screen API it is enough to enable the return assistant feature. The Gini Pay Bank SDK will
+show the return assistant automatically if the invoice contained payable items and will update the extractions returned
+to your app according to the user's changes.
+
+The ``amountToPay`` extraction is updated to be the sum of items the user decided to pay and also includes discounts or
```
Comment (tastyminerals) [inline ginipaybank/src/doc/source/capture-features.rst:138]: I would stop after `decided to pay`. `It includes discounts and additional charges that might be present on the document`.

File: ginipaybank/src/doc/source/capture-features.rst [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+                handleFileImportError(exception)
+            }
+
+            override fun onCancelled() {
+                // Handle cancellation.
+            }
+        })
+    }
+
+Return Assistant
```
Comment (tastyminerals) [inline ginipaybank/src/doc/source/capture-features.rst:119]: I remember Tom wanted us to use a different name but I forgot which one :D Is it still a thing? Maybe we need to ask Anne.

File: ginipaybank/src/doc/source/capture-features.rst [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+
+This is the entry point for the return assistant. It displays the line items extracted from an invoice and their total
+price. The user can deselect line items which should not be paid for and also edit the quantity, price or description of
+each line item. The total price is always updated to include only the selected line items.
+
+The returned extractions in the ``DigitalInvoiceFragmentListener.onPayInvoice()`` are updated to include the user's
+modifications:
+
+* ``amountToPay`` in the specific extractions is updated to contain the sum of the selected line items' prices,
+* the line items in the compound extractions are also updated according to the user's modifications.
```
Comment (tastyminerals) [inline ginipaybank/src/doc/source/capture-features.rst:181]: Should be a separate sentence.

File: ginipaybank/src/main/java/net/gini/pay/bank/capture/digitalinvoice/DigitalInvoiceFragment.kt [hunk 6]
```diff
@@ -45,7 +45,8 @@ private const val TAG_WHAT_IS_THIS_DIALOG = "TAG_WHAT_IS_THIS_DIALOG"
  * - the line items are updated according to the user's modifications.
  *
  * You should show the `DigitalInvoiceFragment` when the
- * [AnalysisFragmentListener.onProceedToReturnAssistant()] is called.
+ * [AnalysisFragmentListener.onExtractionsAvailable()] is called and you have validated the compound extractions using the
```
Comment (tastyminerals) [inline ginipaybank/src/main/java/net/gini/pay/bank/capture/digitalinvoice/DigitalInvoiceFragment.kt:48]: `and you have validated` to `when you have validated`?

File: ginipaybank/src/main/java/net/gini/pay/bank/capture/digitalinvoice/onboarding/DigitalInvoiceOnboardingFragment.kt [hunk 7]
```diff
@@ -18,9 +21,23 @@ import net.gini.pay.bank.databinding.GpbFragmentDigitalInvoiceOnboardingBinding
...
+ *
+ * A [DigitalInvoiceOnboardingFragmentListener] instance must be available before the `DigitalInvoiceOnboardingFragment` is attached to an activity. Failing
+ * to do so will throw an exception. The listener instance can be provided either implicitly by making the hosting Activity implement the
+ * [DigitalInvoiceOnboardingFragmentListener] interface or explicitly by setting the listener using [DigitalInvoiceOnboardingFragment.listener].
+ *
+ * Your Activity is automatically set as the listener in [DigitalInvoiceOnboardingFragment.onCreate()].
+ *
+ * ### Customizing the Digital Invoice Onboarding Screen
+ *
+ * TODO: PPL-14: Customization guide for return assistant - Android
```
Comment (tastyminerals) [inline ginipaybank/src/main/java/net/gini/pay/bank/capture/digitalinvoice/onboarding/DigitalInvoiceOnboardingFragment.kt:40]: still relevant TODO?

Comment (a-szotyori) [reply ginipaybank/src/main/java/net/gini/pay/bank/capture/digitalinvoice/onboarding/DigitalInvoiceOnboardingFragment.kt:40]: Yes 🙂. Ticket is currently parked and will be resumed in the future.

File: ginipaybank/src/main/java/net/gini/pay/bank/capture/digitalinvoice/DigitalInvoiceFragment.kt (hunk 6 above)
Comment (a-szotyori) [reply ginipaybank/src/main/java/net/gini/pay/bank/capture/digitalinvoice/DigitalInvoiceFragment.kt:48]: What do you think about this?
```
Before showing the `DigitalInvoiceFragment` you should validate the compound extractions 
using the [LineItemsValidator]. These extractions are 
returned in the [AnalysisFragmentListener.onExtractionsAvailable()] 
listener method.
```

**Review by tastyminerals — APPROVED** [review]

**Pull request merged**